lib/generators/has_friendship/has_friendship_generator.rb

```diff
@@ -9,7 +9,10 @@ def self.source_root
   end
 
   def self.next_migration_number(path)
-    Time.now.utc.strftime("%Y%m%d%H%M%S")
+    next_num = Time.now.utc.strftime("%Y%m%d%H%M%S").to_i
+    current_num = current_migration_number(path)
+    next_num += (current_num - next_num + 1) if current_num >= next_num
```

Comment:
Thanks for the clarification. How about this:

```rb
next_num = current_num + 1 if current_num == next_num
```

Our solution to avoid overlapping migration timestamp is to bump the next migration timestamp by 1 if it is equal to the `current_num`. `current_num` will never be greater than `next_num`. I feel that the above line will do.

If you really think about it, `x += (y - x + 1)` <-> `x = x + (y - x + 1)` <-> `x = y + 1`. Please advise.